Add exception and divide function

thinkjin99 opened the pull request:
예외 처리를 추가하고 함수를 기능 별로 쪼개는 작업을 진행했습니다.
기존과 달리 S3 연결과 메인 -> 모델 api 호출 부분을 분리했습니다.

thinkjin99 committed:
Add exception and divide function

예외 처리를 추가하고 함수를 기능 별로 쪼개는 작업을 진행했습니다.
기존과 달리 S3 연결과 메인 -> 모델 api 호출 부분을 분리했습니다.

thinkjin99 committed:
Fix None iteration bug

스크래핑할 요소가 없음에도 스크래핑을 실시하며 생기는 에러 수정
valid_urls가 존재하는 경우에만 상세 스크래핑을 실시하는 방식으로 로직 수정
스크래핑 코드에서 업로드와 모델 서버로의 전송 한번에 돌아가게 설정

thinkjin99 committed:
Fix None iteration bug

스크래핑할 요소가 없음에도 스크래핑을 실시하며 생기는 에러 수정
valid_urls가 존재하는 경우에만 상세 스크래핑을 실시하는 방식으로 로직 수정
스크래핑 코드에서 업로드와 모델 서버로의 전송 한번에 돌아가게 설정

thinkjin99 commented:
추가적으로 비슷한 버그가 하나 더 존재하는데 풀 리퀘를 하나 더 열기는 비효율적이라 판단해 여기서 추가적인 수정하고 푸시하겠습니다.

kkanggu commented on scrap_blind.py:
```diff
@@ -205,15 +205,17 @@ async def run():
+                control_s3.upload_json(str(FILE_PATH))
+                control_s3.send_json_to_model()
+            
+            else:
+                logger.info("Not enough new articles")
```
S3 -> Model 데이터 전달시 최근 파일 N개를 병합하여 보내는 것으로 기억하는데, 이렇게 새로운 게시글이 없다면 2시간보다 이전의 데이터가 전달이 되지는 않나요?

kkanggu commented on scrap_blind.py:
```diff
@@ -205,15 +205,17 @@ async def run():
-        else:
-            logger.info("Not enough new articles")
+
+        if len(urls) > 0:
+            infos = await get_all_aritcle(urls)
+            #when new article is morethan 50
+            if len(infos) > 50:
+                create_json(infos)
+                control_s3.upload_json(str(FILE_PATH))
+                control_s3.send_json_to_model()
+            
+            else:
+                logger.info("Not enough new articles")
```
S3 -> Model 데이터 전달시 최근 파일 N개를 병합하여 보내는 것으로 기억하는데, 이렇게 새로운 게시글이 없다면 2시간보다 이전의 데이터가 전달이 되지는 않나요?

thinkjin99 replied:
이 부분 지금 전반적으로 구조를 변경하고 있습니다. 원래 스크래핑 스크립트에서 한번에 진행할 생각을 하고 있었는데, 로깅도 쉽지 않고 에러 처리도 어려워서 작업을 쪼개고 있습니다. 현재 진행하는 방향은 
스크래핑 진행 -> json 생성 -> 모델 서버가 원하는 형태로 가공 -> 모델 용 json 생성 -> 모델 용 json을 모델 api에 전달 -> 모델 api로 부터 응답 받은 키워드 json 생성 -> 다시 키워드 json을 서비스 api로 전달

thinkjin99 committed:
Fix bug about date

블라블라의 광고 글은 2022.09.07의 형태로 다른 게시글과 다른 날짜 형식으로
업로드 됨. 이에따라 파싱에러가 생기는 부분 수정함.
그 외 last_scrapped_.json이 비어있는 경우 생기는 버그 역시 조건문을 추가해 수정.

thinkjin99 committed:
Refacotr logger

로거를 기존처럼 마구잡이로 사용시 로거가 여러개 생성되면서 로그가 여러개 찍히는 현상 발생
핸들러가 존재하지 않는 경우에만 로그 인스턴스 생성하도록 제약 줌

thinkjin99 committed:
Merge branch 'feature' of https://github.com/Soongsil-Developers/22sdc-issue-load into feature

최신 코드 상황 반영해 병합

thinkjin99 committed:
Refactor function flow logic

함수에서 json을 생성하고 이를 json 폴더에 보관한다.
기존과 달리 json 데이터를 생성하자마자 전송하지 않는다.

thinkjin99 committed:
Add script for sending json to model api

모델 서버로 json을 전송한다.

kkanggu commented on scrap_blind.py:
```diff
@@ -205,15 +221,17 @@ async def run():
-            logger.info("Not enough new articles")
+        if urls:
+            #drop advertise articles
+            infos = [info for info in await get_all_aritcle(urls) if info]
```
함수 이름이 `article`이 아닌 `aritcle`로 오타가 존재합니다. 수정 부탁드립니다...

thinkjin99 committed:
Add send json to service api

서비스 api로 모델에게서 전달 받은 키워드 json을 전송합니다.

thinkjin99 committed:
Fix aritcle to article
